src/views/tables/OrdersTable/CancelOrClearAllOrdersButton.tsx

```diff
@@ -18,45 +20,38 @@ type ElementProps = {
 };
 
 export const CancelOrClearAllOrdersButton = ({ marketId }: ElementProps) => {
-  const { cancelAllOrders } = useSubaccount();
+  const stringGetter = useStringGetter();
   const dispatch = useDispatch();
+  const { cancelAllOrders } = useSubaccount();
   const hasCancelableOrders = useParameterizedSelector(calculateHasCancelableOrders, marketId);
 
-  const onCancelAll = useCallback(() => {
-    cancelAllOrders(marketId);
-  }, [cancelAllOrders, marketId]);
-
-  const onClearAll = useCallback(() => {
-    dispatch(clearAllOrders(marketId));
-  }, [dispatch, marketId]);
+  const onClearOrCancelAll = useCallback(() => {
+    if (hasCancelableOrders) {
+      cancelAllOrders(marketId);
+    } else {
+      dispatch(clearAllOrders(marketId));
+    }
+  }, [cancelAllOrders, dispatch, hasCancelableOrders, marketId]);
 
-  // TODO(@aforaleka): Localize strings
   // TODO(@aforaleka): add cancel all confirmation dialog
-  return hasCancelableOrders ? (
-    <$CancelAllButton action={ButtonAction.Primary} size={ButtonSize.XSmall} onClick={onCancelAll}>
-      Cancel All
-    </$CancelAllButton>
-  ) : (
-    <$ActionTextButton action={ButtonAction.Primary} size={ButtonSize.XSmall} onClick={onClearAll}>
-      Clear All
+  return (
+    <$ActionTextButton
+      action={ButtonAction.Primary}
+      size={ButtonSize.XSmall}
+      onClick={onClearOrCancelAll}
+      isHighlighted={hasCancelableOrders}
```

Comment:
hmm is it intentional that we don't want "Clear All" to also be purple? I sorta like having them as both Purple, otherwise it doesn't seem very obvious that it's clickable
<img width="222" alt="Screenshot 2024-09-17 at 1 05 25 PM" src="https://github.com/user-attachments/assets/b480f61d-ef13-4e1f-b52e-bced9f8952f1">

Reply:
i think mostly to distinguish it from 'cancel all' and same with the faded X for dismiss -- but will sync with product/design for feedback!